packages/tasit-action/src/TasitAction.js

```diff
@@ -107,12 +117,51 @@ class TransactionSubscription extends Subscription {
     this.#provider = provider;
   }
 
-  #addConfirmationListener = async listener => {
+  on = (eventName, listener) => {
+    this.#addListener(eventName, listener, false);
+  };
+
+  once = (eventName, listener) => {
+    this.#addListener(eventName, listener, true);
+  };
+
+  getEventsTimeout = () => {
+    return this.#timeout;
+  };
+
+  setEventsTimeout = timeout => {
+    this.#timeout = timeout;
+  };
+
+  #addListener = (eventName, listener, once) => {
+    const triggers = ["confirmation", "error"];
+
+    if (!triggers.includes(eventName))
+      throw new Error(`Invalid listener trigger, use: [${triggers}]`);
+
+    if (eventName === "error" && once)
+      throw new Error(`Use on() function to subscribe an error event.`);
+
+    if (!listener || typeof listener !== "function")
+      throw new Error(`Cannot listen without a function`);
+
+    if (eventName === "error") {
+      this._addErrorListener(listener);
+    } else if (eventName === "confirmation") {
+      this.#addConfirmationListener(listener, once);
```

Comment:
The https://github.com/tc39/proposal-class-fields don't support protected methods/fields, only private (subclass can't access them), so I'm using underscore to indicate protected methods.
Do you have any suggestion to do it better?

Reply:
Got it - that sounds perfect!

Reply:
And just to confirm, in this case `addConfirmationListener` needs to be private whereas `addErrorListener` needs to be protected?

Reply:
The visibility difference between these methods is because `#addConfirmationListener` lives on `TransactionSubscription` and `_addErrorListener` inside `Subscription` superclass.
If I change `_addErrorListener` to private (#) the `TransactionSubscription` subclass can't access it.